src/OpenTelemetry/Trace/Configuration/ActivityProcessorPipelineBuilder.cs

```diff
@@ -0,0 +1,126 @@
+﻿// <copyright file="ActivityProcessorPipelineBuilder.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Trace.Export;
+
+namespace OpenTelemetry.Trace.Configuration
+{
+    /// <summary>
+    /// Configures exporting pipeline with chains of processors and exporter.
+    /// </summary>
+    public class ActivityProcessorPipelineBuilder
+    {
+        private Func<ActivityExporter, ActivityProcessor> lastProcessorFactory;
+        private List<Func<ActivityProcessor, ActivityProcessor>> processorChain;
+
+        internal ActivityProcessorPipelineBuilder()
+        {
+        }
+
+        internal ActivityExporter Exporter { get; private set; }
+
+        internal List<ActivityProcessor> Processors { get; private set; }
+
+        /// <summary>
+        /// Adds chained processor to the pipeline. Processors are executed in the order they were added.
+        /// </summary>
+        /// <param name="processorFactory">Function that creates processor from the next one.</param>
+        /// <returns>Returns <see cref="ActivityProcessorPipelineBuilder"/>.</returns>
+        public ActivityProcessorPipelineBuilder AddProcessor(Func<ActivityProcessor, ActivityProcessor> processorFactory)
+        {
+            if (processorFactory == null)
+            {
+                throw new ArgumentNullException(nameof(processorFactory));
+            }
+
+            if (this.processorChain == null)
+            {
+                this.processorChain = new List<Func<ActivityProcessor, ActivityProcessor>>();
+            }
+
+            this.processorChain.Add(processorFactory);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Configures last processor that invokes exporter. When not set, <see cref="SimpleActivityProcessor"/> is used.
+        /// </summary>
+        /// <param name="processorFactory">Factory that creates exporting processor from the exporter.</param>
+        /// <returns>Returns <see cref="ActivityProcessorPipelineBuilder"/>.</returns>
+        public ActivityProcessorPipelineBuilder SetExportingProcessor(Func<ActivityExporter, ActivityProcessor> processorFactory)
```

Comment:
Could we determine the 'exporting' processor is the last processor in the list instead of explicitly setting it?

Reply:
This is the same behavior as current SpanProcessorPipelineBuilder - I'd defer any changes to current design in subsequent PRs to unblock progress on this PR.